### PR TITLE
Introduction to GUI using Tkinter in Python

### LINKED ISSUE #2259: introduction to GUI using tkinter with python
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Proposed title of article

introduction to GUI using tkinter with python.

### Introduction paragraph (2-3 paragraphs):
The term GUI (graphic user interface) refers to a desktop application that allows you to communicate with computers and improves the experience of giving commands (command-line input) to your code. They are used in desktop computers, notebook computers, and other electronic devices to perform a variety of tasks.
Tkinter is a platform that is included with Python.
### Key takeaways: 
In this tutorial, programmers will learn and know
- What are Widgets?
- How to create a simple graphic user interface
- Inserting of menu keys and background colour to make it more user friendly.
### References: 
python GUI programming with tkinter by Alan D. Moore.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by majiezra**
This fixes #2259 
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by majiezra** [comment]
This fixes issue #2502

**Review by geoffrey45 — CHANGES_REQUESTED** [review]
Hello @majiezra Thank you for submitting this article for review.

This is a great topic. 

Your article has some few issues though. It has a lot of code lacking explanation. Please look at the articles already published at https://www.section.io/engineering-education/

I've left some comments below to help you with the issues that need fixing. Please check for issues mentioned in a comment throughout your tutorial.

File: content/articles/introduction-to-gui-using-tkinter-in-python/index.md [hunk 1]
```diff
@@ -0,0 +1,329 @@
+
+![heroimage](/engineering-education/introduction-to-gui-using-tkinter-in-python/hero.jpg)
```
[the remark this answers is not in the document]
Comment (majiezra) [reply content/articles/introduction-to-gui-using-tkinter-in-python/index.md:2]: @geoffrey45  Good day
thank you very much for the review, I will make the corrections you just highlighted, concerning the hero image should I fix a new image or leave it empty??
thank you once again.

[the remark this answers is not in the document]
Comment (geoffrey45) [reply content/articles/introduction-to-gui-using-tkinter-in-python/index.md:2]: @majiezra Just the text in the md file. Don't delete the image life

[the remark this answers is not in the document]
Comment (majiezra) [reply content/articles/introduction-to-gui-using-tkinter-in-python/index.md:2]: okay. thank you

[the remark this answers is not in the document]
Comment (majiezra) [reply content/articles/introduction-to-gui-using-tkinter-in-python/index.md:2]: @geoffrey45 I don't know if you got my message on slack.
thanks

**Comment by majiezra** [comment]
@geoffrey45  Good evening!
I have made the corrections.
Thank you

File: content/articles/introduction-to-gui-using-tkinter-in-python/index.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+Let's start by importing Tkinter.
+
+```Python
+from tkinter import *
+from tkinter.ttk import Combobox
+import random
+```
+
+The `tkinter.ttk` module gives the user access to the Tkinter widget set and allows the user to import `Combobox`. A combobox is a graphical user interface component that combines a drop-down box, list box, or allows the user to select input.
```
Comment (geoffrey45) [inline content/articles/introduction-to-gui-using-tkinter-in-python/index.md:64]: You have not told us where to put the code you have provided. Please provide instructions on this. For example, `Create a new file passgen.py and add the code below`

File: content/articles/introduction-to-gui-using-tkinter-in-python/index.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+```Python
+screen = Tk()
+screen.title("Password Generator")
+screen.geometry('600x400')
+screen.configure(background ="bisque")
+``` 
+
+A root window is created using the `Tk` class. The `Tk()` function assists in the creation of this GUI window, and provides numerous options such as setting the title and the geometry of the GUI window. The `geometry()` method is one of the many methods provided by Tkinter. We'll be working on the backdrop color of the Tkinter GUI window in this line of code.
```
Comment (geoffrey45) [inline content/articles/introduction-to-gui-using-tkinter-in-python/index.md:73]: When you say `We'll be working on the backdrop color of the Tkinter GUI window in this line of code`, which line are you talking of? There are lots of lines that can be interpreted as `this line`.

Try something like `We change the background color using the screen.configure() method.`

File: content/articles/introduction-to-gui-using-tkinter-in-python/index.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+   global sc1
+   sc1.set("")
+   passw=""
+   length=int(c1.get())
+   lowercase='abcdefghijklmnopqrstuvwxyz'
+   uppercase='ABCDEFGHIJKLMNOPQRSTUVWXYZ'+lowercase
+   mixs='0123456789'+lowercase+uppercase+'@#$%&*'
+```
+
+The global variable `sc1` is defined in the line code `global sc1`. We used the `.set()` method to set the value of `sc1` to null. We also assign a null value to the `passw` variable. The `length` variable will retain the value of `c1`, and the value type of `c1` will be integer. We then create a lowercase variable that holds a string value. After that, we create the `uppercase` variable to hold a string value. And at the last line of code, we define the `mixs`.
```
Comment (geoffrey45) [inline content/articles/introduction-to-gui-using-tkinter-in-python/index.md:86]: There are lots of variables in these pieces of code. Please tell us what these variables hold.


Also, there is a differece between `null` and `empty`. Revise that part.

File: content/articles/introduction-to-gui-using-tkinter-in-python/index.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    for i in range(0,length):
+        passw=passw+random.choice(uppercase)
+    sc1.set(passw)
+elif c2.get()=='High Strength':
+    for i in range(0,length):
+        passw=passw+random.choice(mixs)
+    sc1.set(passw)
+```
+
+We define the `if` condition to monitor the strength of the passwords generated using the password generator.
```
Comment (geoffrey45) [inline content/articles/introduction-to-gui-using-tkinter-in-python/index.md:105]: Explain the rules used to determine the password strength.

**Review by geoffrey45 — CHANGES_REQUESTED** [review]
Hello @majiezra 

I am through the prelim review on your article. I have come across a lot of composition issues. Please see the changes I have made to your document.

Also check the attached comments.

File: content/articles/introduction-to-gui-using-tkinter-in-python/index.md (hunk 5 above)
Comment (majiezra) [reply content/articles/introduction-to-gui-using-tkinter-in-python/index.md]: @geoffrey45  Done.
Thank you

**Comment by hectorkambow** [comment]
[Introduction to GUI using Tkinter in Python #2502.pdf](https://github.com/section-engineering-education/engineering-education/files/6697506/Introduction.to.GUI.using.Tkinter.in.Python.2502.pdf)

👋 Good afternoon - Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on plagiarism and what is considered as such.

@majiezra

**Comment by majiezra** [comment]
> [Introduction to GUI using Tkinter in Python #2502.pdf](https://github.com/section-engineering-education/engineering-education/files/6697506/Introduction.to.GUI.using.Tkinter.in.Python.2502.pdf)
> 
> 👋 Good afternoon - Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
> Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
> Be sure to see our resources page to see more info on plagiarism and what is considered as such.
> 
> @majiezra

Good evening @hectorkambow  the corrections have been made.
Thanks

File: content/articles/introduction-to-gui-using-tkinter-in-python/index.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+Then, using x and y coordinate values, we assigned the location of the `c2` `Combobox`. We define `b` as a button and use the generator command to conduct an action when the button is clicked.
+
+Then, we use x and y coordinates to assign the location of the button `b`. The `mainloop()` function is an infinite loop that is used to run the program.
+
+Here is an image of the password generator interface.
+
+![password image](/engineering-education/introduction-to-gui-using-tkinter-in-python/password.png)
+
+Congratulations on finishing this tutorial.
+
```
Comment (geoffrey45) [inline content/articles/introduction-to-gui-using-tkinter-in-python/index.md:153]: @majiezra Please add a link to your code on GitHub.

Comment (majiezra) [reply content/articles/introduction-to-gui-using-tkinter-in-python/index.md:153]: done 
@geoffrey45

**Review by geoffrey45 — APPROVED** [review]

**Comment by ahmadmardeni1** [comment]
Good work gentlemen @majiezra @geoffrey45 
However, there are some points we have to focus on:
- I found major grammar mistakes, please use [Grammarly](https://app.grammarly.com/) or any other tool provided in our docs in your next article. 
- There are a lot of repeated sentences but in different contexts, please be precise, the reader may feel bored.
Go through my changes.

@geoffrey45  Please do the same, we need to improve the future reviews.

**Pull request merged**